docs/design-v2.md

```diff
@@ -1,133 +1,77 @@
 
 ### Custom Resources and Controllers
 
-Mandalay uses 3 different custom resources defined in the `azure-disk-csi` namespace to 
-orchestrate disk management and facilitate pod failover to nodes with attachment replica. A 
-controller for each resource watches for and responds to changes to its custom resource 
-instances.
+The V2 driver uses 3 different custom resources defined in the `azure-disk-csi` namespace to orchestrate disk management and facilitate pod failover to nodes with attachment replica. A controller for each resource watches for and responds to changes to its custom resource instances.
 
 #### `AzDriverNode` Resource and Controller
 
-The `AzDriverNode` custom resource represents a node in the cluster where the Mandalay node 
-plug-in runs. An instance of `AzDriverNode` is created when the node plug-in starts. The node 
-plug-in periodically updates the heartbeat in the `Status` field.
+The `AzDriverNode` custom resource represents a node in the cluster where the V2 node plug-in runs. An instance of `AzDriverNode` is created when the node plug-in starts. The node plug-in periodically updates the heartbeat in the `Status` field.
 
-The controller for `AzDriverNode` runs in the controller plug-in. It is responsible for 
-deleting `AzDriverNode` instances that no longer have corresponding nodes in the cluster.
+The controller for `AzDriverNode` runs in the controller plug-in. It is responsible for deleting `AzDriverNode` instances that no longer have corresponding nodes in the cluster.
 
 #### `AzVolume` Resource and Controller
 
-The `AzVolume` custom resource represents the managed disk of a `PersistentVolume`. The 
-controller for `AzVolume` runs in the controller plug-in. It watches for and reconciles 
-changes in the `AzVolume` instances.
+The `AzVolume` custom resource represents the managed disk of a `PersistentVolume`. The controller for `AzVolume` runs in the controller plug-in. It watches for and reconciles changes in the `AzVolume` instances.
 
-An `AzVolume` instance is created by the `CreateVolume` API to the CSI Controller plug-in. The 
-`AzVolume` controller responds to the new instance by creating a managed disk using the 
-parameters in the referenced `StorageClass`. When disk creation is complete, the controller 
-sets the `.Status.State` field to `Created` or suitable error state on failure. It also adds a 
-finalizer if creation was successful. The `CreateVolume` request completes when the `AzVolume` 
-state is updated.
+An `AzVolume` instance is created by the `CreateVolume` API to the CSI Controller plug-in. The `AzVolume` controller responds to the new instance by creating a managed disk using the parameters in the referenced `StorageClass`. When disk creation is complete, the controller sets the `.Status.State` field to `Created` or suitable error state on failure. It also adds a finalizer if creation was successful. The `CreateVolume` request completes when the `AzVolume` state is updated.
```

Comment:
>It also adds a finalizer if creation was successful.
It adds a finalizer to the instance upon CRI creation.

Reply:
I will drop mention of the finalizers. I don't believe when they are added is critical to understanding the overall design and workflows.